### PR TITLE
[lua/test] Add tc for multiple tensor types in lua filter

### PR DESCRIPTION
**Pull request by anyj0527**
- Add a testcase which covers all tensor types in tensor_filter::lua

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc [hunk 1]
```diff
@@ -891,6 +910,53 @@ end
...
+    ret = sp->invoke (NULL, NULL, data, &input, &output);
+    EXPECT_EQ (ret, 0);
+
+    g_free (model);
+  }
+
+  g_free (input.data);
+  g_free (output.data);
+  sp->close (&prop, &data);
+}
```
Comment (myungjoo) [inline tests/nnstreamer_filter_lua/unittest_filter_lua.cc:958]: Please add negative cases (```${name}_${num}_n```)

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc [hunk 2]
```diff
@@ -891,6 +892,55 @@ end
...
+  GstTensorMemory input, output;
+  GstTensorFilterProperties prop;
+
+  output.size = input.size = sizeof (int64_t) * 1 * 2 * 2 * 1;
+
+  /* alloc input data without alignment */
+  input.data = g_malloc (input.size);
+  output.data = g_malloc (output.size);
+
+  memset (input.data, 0, input.size);
```
Comment (gichan-jang) [inline tests/nnstreamer_filter_lua/unittest_filter_lua.cc:911]: How about using g_malloc0?

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc [hunk 3]
```diff
@@ -891,6 +892,55 @@ end
...
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("lua");
+  EXPECT_NE (sp, nullptr);
+
+  for (int i = 0; i < _NNS_END; ++i) {
+    tensor_type ttype = (tensor_type) i;
```
Comment (gichan-jang) [inline tests/nnstreamer_filter_lua/unittest_filter_lua.cc:917]: `i` is duplicated.

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc (hunk 2 above)
Comment (anyj0527) [reply tests/nnstreamer_filter_lua/unittest_filter_lua.cc]: g_malloc0 seems good. Thanks!

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc (hunk 1 above)
Comment (anyj0527) [reply tests/nnstreamer_filter_lua/unittest_filter_lua.cc:974]: I added a negative case for using invalid data type.

File: tests/nnstreamer_filter_lua/unittest_filter_lua.cc [hunk 4]
```diff
@@ -891,6 +892,87 @@ end
...
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("lua");
+  EXPECT_NE (sp, nullptr);
+
+  /**
+   * Invalid data type `uint128`. Type should be one of
+   * { float32, float64, int64, uint64, int32, uint32, int16, uint16, int8, uint8 }
+   */
+  const char *invalid_data_type = "uint128";
+  const gchar *model = g_strdup_printf ("\
```
Comment (jaeyun-jung) [inline tests/nnstreamer_filter_lua/unittest_filter_lua.cc:912]: check returned type (char to const char)
g_strdup_printf() returns newly allocated string, need free function.
I recommend to add macro to get model string,
```
#define make_lua_model(dtype) "utTensorsInfo={num=1,dim={{1,2,2,1},},type={" dtype ",}} ...
....

in dataType00_n()
const gchar model[] = make_lua_model ("uint128");

in dataType01()
const gchar model[] = make_lua_model (gst_tensor_get_type_string (ttype));
```

Comment (anyj0527) [reply tests/nnstreamer_filter_lua/unittest_filter_lua.cc]: Instead of macro, I updated the code to call `g_free` for allocated strings. Thanks, PTAL!

**Review by jaeyun-jung — APPROVED** [review]

**Review by myungjoo — APPROVED** [review]

**Pull request merged**